pkg/networkmanager/network-main.jsx

```diff
@@ -138,6 +138,46 @@
 
     const actions = privileged && (
         <>
+            <Button onClick={ async () => {
+                const key = await cockpit.script("wg genkey");
+
+                const client = cockpit.dbus("org.freedesktop.NetworkManager", { superuser: 'try' });
+                try {
+                    const reply = await client.call("/org/freedesktop/NetworkManager/Settings",
+                                                    "org.freedesktop.NetworkManager.Settings",
+                                                    "AddConnection",
+                                                    [
+                                                        {
+                                                            connection: {
+                                                                id: { t: 's', v: 'connection-wg0' },
+                                                                type: { t: 's', v: 'wireguard' },
+                                                                'interface-name': { t: 's', v: 'wg0' }
+                                                            },
+                                                            wireguard: {
+                                                                'private-key': { t: 's', v: key },
+                                                                'listen-port': { t: 'u', v: 51820 },
+                                                            },
+                                                            ipv4: {
+                                                                'address-data': {
+                                                                    t: 'aa{sv}',
+                                                                    v: [{ address: { t: 's', v: '10.0.0.1' }, prefix: { t: 'u', v: 24 } }]
+                                                                },
+                                                                method: {
+                                                                    t: 's',
+                                                                    v: 'manual'
+                                                                }
+                                                            }
+                                                        }
+                                                    ]
+                    );
+                    console.log('reply => ', reply);
+                } catch (e) {
+                    console.log('error => ', e);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230626-161008-14729dcc-fedora-38-devel-networking//Coverage/lcov/github-pr.diff.gcov.html)